src/packages/version-check/tests/version-check.test.ts

```diff
@@ -0,0 +1,737 @@
+      );
+    });
+
+    it("message contains the packageName", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      vc.log();
+
+      assert.strictEqual(
+        message.indexOf(vc._config.packageName) >= 0,
+        true,
+        "Message does not contain the upgradeType."
+      );
+    });
+    it("message contains the currentVersion", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      vc.log();
+
+      assert.strictEqual(
+        message.indexOf(vc._currentVersion) >= 0,
+        true,
+        "Message does not contain the upgradeType."
+      );
+    });
+    it("message contains the latestVersion", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      vc.log();
+
+      assert.strictEqual(
+        message.indexOf(vc._config.latestVersion) >= 0,
+        true,
+        "Message does not contain the upgradeType."
+      );
+    });
+    it("process.stdout.columns === -1", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      const oldColumns = process.stdout.columns;
+      process.stdout.columns = -1;
+
+      vc.log();
+      process.stdout.columns = oldColumns;
+      assert.strictEqual(message.length > 0, true, "Message did not log.");
+    });
+    it("process.stdout.columns === null", () => {
+      vc = new VersionCheck(testVersion, testConfig, testLogger);
+      const oldColumns = process.stdout.columns;
+      process.stdout.columns = 0;
+
+      vc.log();
+      process.stdout.columns = oldColumns;
+      assert.strictEqual(message.length > 0, true, "Message did not log.");
+    });
+  });
+
+  describe("getLatestVersion/_fetchLatest", () => {
+    let api;
+    const apiResponse = "1.0.0";
+    const apiSettings = {
+      port: 4000
+    };
+    const sleep = milliseconds => {
+      return new Promise(resolve => setTimeout(resolve, milliseconds));
+    };
+
+    beforeEach(() => {
+      api = http2.createServer();
+      api.on("error", err => console.error(err));
+
+      api.on("stream", async (stream, headers) => {
+        const path = headers[":path"];
+        const method = headers[":method"];
+
+        if (method !== "GET") {
+          stream.respond({
+            ":status": 404
+          });
+          stream.end();
+        }
+        switch (path) {
+          case "/version?package=ganache":
+            if (stream.closed) return;
+            stream.respond({
+              ":status": 200
+            });
+            stream.write(apiResponse);
+            stream.end();
+            break;
+          default:
+            await sleep(10);
+            stream.respond({
+              ":status": 404
+            });
+            stream.end();
+            break;
+        }
+      });
+
+      api.listen(apiSettings.port);
+      vc = new VersionCheck(
+        testVersion,
+        {
+          enabled: true,
+          url: "http://localhost:" + apiSettings.port
+        },
```

Comment:
This could be a good opportunity to use nock - rather than setting up an actually http server.

Reply:
nock does not support http2